src/Microsoft.OData.Edm/Csdl/Serialization/EdmModelCsdlSerializationVisitor.cs

```diff
@@ -653,7 +656,7 @@ private void ProcessReferentialConstraint(IEdmReferentialConstraint element)
             }
         }
 
-        private void ProcessFacets(IEdmTypeReference element, bool inlineType)
+        private void ProcessFacets(IEdmTypeReference element, bool inlineType, bool nullableAttributeRequiredForCollection)

```

Comment:
>nullableAttributeRequiredForCollection [](http://example.com/codeflow?start=84&length=38)

So this gets set true for structural property facets, and false otherwise?  Can we just call this something like "isStructuralPropertyDeclaration"?

Reply:
the call "chain" is 
    EdmModelCsdlSerializationVisitor.{ProcessStructuralProperty|ProcessTerm|ProcessCollectionType|...)
    EdmModelCsdlSerializationVisitor.ProcessFacets -> 
    EdmModelCsdlSchemaWriter.WriteNullableAttribute 
    
and  EdmModelCsdlSchemaWriter is abstract (one concrete subclass for XML, one for JSON)
    
And yes, it is true that only ProcessStructuralProperty is passing in true. But that doesn't describe the behavior of ProcessFacets which is : instruct the writer to write the property for a collection type (-reference) even if it is false (which otherwise dosn't get written). The name is based on what the method does, not on how and when it is called. 

But I don't have a strong opinion. Happy to change it if you prefer.

Reply:
Let's discuss --  my concern is that, as I understand it, a component that should be format-agnostic (EdmModelCsdlSerializationVistor) would be making the decision of whether to write the nullable attribute or not based on the format (write for xml, not for json).  ideally, we wouldn't be making format-specific decisions in these common classes, they would all be made in the format-specific implementations.